src/runtimeMain/kotlin/com/epam/drill/test/agent/http/HttpClient.kt

```diff
@@ -0,0 +1,84 @@
+import com.epam.drill.test.agent.instrumentation.testing.logging.*
+import java.net.*
+import java.security.*
+import java.security.cert.*
+import javax.net.ssl.*
+
+object HttpClient {
+    val logger = Logging.logger(HttpClient::class.java.name)
+
+    init {
+        // Create a trust manager that does not validate certificate chains
+        val trustAllCerts = arrayOf<TrustManager>(
+            object : X509TrustManager {
+                @Throws(CertificateException::class)
+                override fun checkClientTrusted(
+                    chain: Array<X509Certificate>,
+                    authType: String,
+                ) {
+                }
+
+                @Throws(CertificateException::class)
+                override fun checkServerTrusted(
+                    chain: Array<X509Certificate>,
+                    authType: String,
+                ) {
+                }
+
+                override fun getAcceptedIssuers(): Array<X509Certificate> {
+                    return arrayOf()
+                }
+            }
+        )
+        // Install the all-trusting trust manager
+        val sslContext = SSLContext.getInstance("SSL")
+        sslContext.init(null, trustAllCerts, SecureRandom())
+        // Create an ssl socket factory with our all-trusting manager
+        val sslSocketFactory: SSLSocketFactory = sslContext.socketFactory
+        HttpsURLConnection.setDefaultSSLSocketFactory(sslSocketFactory)
+        HttpsURLConnection.setDefaultHostnameVerifier { _, _ -> true }
+    }
+
+    fun request(url: String, block: HttpUrlConnectionBuilder.() -> Unit = { }): HttpResponse = runCatching {
+        val urlConnection = HttpUrlConnectionBuilder(url).apply(block).build()
+        urlConnection.requestMethod
+        urlConnection.connect()
+        return HttpResponse(
+            urlConnection.responseCode,
+            urlConnection.headerFields.mapValues { it.value.first() },
+            urlConnection.inputStream.reader().readText()
+        )
+    }.onFailure { logger.error { "Can't get response from $url" } }.getOrElse { HttpResponse(500) }
+}
+
+data class HttpUrlConnectionBuilder(
+    var url: String,
+    val queryParams: MutableMap<String, String> = mutableMapOf(),
+    var method: HttpMethod = HttpMethod.GET,
+    val headers: MutableMap<String, String> = mutableMapOf(),
+    var body: String = "",
+) {
+
+    fun build(): HttpURLConnection = (URL(url + queryParamString()).openConnection() as HttpURLConnection).apply {
+        requestMethod = method.name
+        doOutput = true
+        setRequestProperty("Content-Type", "application/json")
+        headers.forEach { (key, value) ->
+            setRequestProperty(key, value)
+        }
+        if (body.isNotBlank()) {
+            outputStream.use { it.write(body.toByteArray()) }
+        }
+    }
+
+    private fun queryParamString() = queryParams.entries.joinToString(separator = "&", prefix = "?") {
+        "${it.key}=${it.value}"
+    }
```

Comment:
Are there cases when the parameters may contain invalid characters (UTF) or spaces ?
If yes, then we need to encode them.
```suggestion
    private fun queryParamString() = queryParams.entries.joinToString(separator = "&", prefix = "?") {
        "${it.key.urlEncode()}=${it.value.urlEncode()}"
    }
```